packages/react-next/src/components/SwatchColorPicker/SwatchColorPicker.base.tsx

```diff
@@ -16,361 +8,341 @@ import {
-  public static defaultProps = {
-    cellShape: 'circle',
-    disabled: false,
-    shouldFocusCircularNavigate: true,
-    cellMargin: 10,
-  } as ISwatchColorPickerProps;
-
-  private _id: string;
-  private _cellFocused: boolean;
-
-  private navigationIdleTimeoutId: number | undefined;
-  private isNavigationIdle: boolean;
-  private readonly navigationIdleDelay: number = 250 /* ms */;
-  private async: Async;
-
-  // Add an index to each color cells. Memoizes this so that color cells do not re-render on every update.
-  private _getItemsWithIndex = memoizeFunction((items: IColorCellProps[]) => {
-    return items.map((item, index) => {
-      return { ...item, index: index };
+function useDebugWarnings(props: ISwatchColorPickerProps) {
+  if (process.env.NODE_ENV !== 'production') {
+    // eslint-disable-next-line react-hooks/rules-of-hooks -- build-time conditional
+    useWarnings({
+      name: COMPONENT_NAME,
+      props,
+      mutuallyExclusive: { focusOnHover: 'onHover' },
+      deprecations: { positionInSet: 'ariaPosInSet', setSize: 'ariaSetSize' },
     });
-  });
-
-  constructor(props: ISwatchColorPickerProps) {
-    super(props);
-
-    this._id = props.id || getId('swatchColorPicker');
-
-    if (process.env.NODE_ENV !== 'production') {
-      warnMutuallyExclusive(COMPONENT_NAME, props, {
-        focusOnHover: 'onHover',
-      });
-
-      warnConditionallyRequiredProps(
-        COMPONENT_NAME,
-        props,
-        ['focusOnHover'],
-        'mouseLeaveParentSelector',
-        !!props.mouseLeaveParentSelector,
-      );
-
-      warnDeprecations(COMPONENT_NAME, props, {
-        positionInSet: 'ariaPosInSet',
-        setSize: 'ariaSetSize',
-      });
-    }
-
-    this.isNavigationIdle = true;
-    this.async = new Async(this);
-
-    let selectedIndex: number | undefined;
-    if (props.selectedId) {
-      selectedIndex = this._getSelectedIndex(props.colorCells, props.selectedId);
-    }
-
-    this.state = {
-      selectedIndex,
-    };
-  }
-
-  public UNSAFE_componentWillReceiveProps(newProps: ISwatchColorPickerProps): void {
-    if (newProps.selectedId !== undefined) {
-      this.setState({
-        selectedIndex: this._getSelectedIndex(newProps.colorCells, newProps.selectedId),
-      });
-    }
   }
+}
 
-  public componentWillUnmount() {
-    if (this.props.onCellFocused && this._cellFocused) {
-      this._cellFocused = false;
-      this.props.onCellFocused();
-    }
-    this.async.dispose();
-  }
+export const SwatchColorPickerBase = React.forwardRef<HTMLElement, ISwatchColorPickerProps>((props, ref) => {
+  const defaultId = useId('swatchColorPicker');
+  const id = props.id || defaultId;
+
+  const internalState = useConst<ISwatchColorPickerState>(() => ({
+    isNavigationIdle: true,
+    cellFocused: false,
+    navigationIdleTimeoutId: undefined,
+    navigationIdleDelay: 250,
+  }));
```

Comment:
Small thing but you don't need a wrapping function here.
```suggestion
  const internalState = useConst<ISwatchColorPickerState>({
    isNavigationIdle: true,
    cellFocused: false,
    navigationIdleTimeoutId: undefined,
    navigationIdleDelay: 250,
  });
```